application/src/main/java/org/togetherjava/tjbot/features/roleapplication/ApplicationCreateCommand.java

```diff
@@ -0,0 +1,290 @@
+            return;
+        }
+
+        long incorrectArgsCount = getIncorrectRoleArgsCount(optionMappings);
+        if (incorrectArgsCount > 0) {
+            event.reply("Missing information for %d roles.".formatted(incorrectArgsCount))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        sendMenu(event);
+    }
+
+    @Override
+    public void onStringSelectSelection(StringSelectInteractionEvent event, List<String> args) {
+        SelectOption selectOption = event.getSelectedOptions().getFirst();
+        Member member = event.getMember();
+
+        if (selectOption == null || member == null) {
+            return;
+        }
+
+        long remainingMinutes = applicationApplyHandler.getMemberCooldownMinutes(member);
+        String correctMinutesWord = selectWordFromCount(remainingMinutes, "minute", "minutes");
+
+        if (remainingMinutes > 0) {
+            event
+                .reply("Please wait %d %s before sending a new application form."
+                    .formatted(remainingMinutes, correctMinutesWord))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        TextInput body = TextInput
+            .create(generateComponentId(event.getUser().getId()),
+                    roleApplicationSystemConfig.defaultQuestion(), TextInputStyle.PARAGRAPH)
+            .setRequired(true)
+            .setRequiredRange(roleApplicationSystemConfig.minimumAnswerLength(),
+                    roleApplicationSystemConfig.maximumAnswerLength())
+            .setPlaceholder("Enter your answer here")
+            .build();
+
+        EmojiUnion emoji = selectOption.getEmoji();
+        String roleDisplayName;
+
+        if (emoji == null) {
+            roleDisplayName = selectOption.getLabel();
+        } else {
+            roleDisplayName = "%s %s".formatted(emoji.getFormatted(), selectOption.getLabel());
+        }
+
+        Modal modal = Modal
+            .create(generateComponentId(event.getUser().getId(), roleDisplayName),
+                    String.format("Application form - %s", selectOption.getLabel()))
+            .addActionRow(ActionRow.of(body).getComponents())
+            .build();
+
+        event.replyModal(modal).queue();
+    }
+
+    /**
+     * Selects and returns the appropriate singular or plural form of a word based on the given
+     * count.
+     *
+     * @param singularForm the word in its singular form
+     * @param pluralForm the word in its plural form
+     * @param count the number used to determine whether to return the singular or plural form
+     * @return the singular form if count equals 1, otherwise the plural form
+     */
+    private String selectWordFromCount(final Number count, final String singularForm,
+            final String pluralForm) {
+        if (count.intValue() == 1) {
+            return singularForm;
+        }
+        return pluralForm;
+    }
```

Comment:
i think this is a bit overengineered. can remove that helper (as its also only used once) and instead make the call-site a bit easier to read by inlining:
```java
String minutesText = remainingMinutes == 1 ? "minute", "minutes";